graspologic/nominate/spectralVN.py

```diff
@@ -0,0 +1,316 @@
+        self.input_graph = input_graph
+
+    @staticmethod
+    def _make_2d(arr: np.ndarray) -> np.ndarray:
+        # ensures arr is two or less dimensions.
+        # if 1d, adds unique at each index on
+        # the second dimension.
+        if np.ndim(arr) == 1 or arr.shape[1] == 1:
+            arr = arr.reshape(-1, 1)
+            arr = np.concatenate((arr, np.arange(arr.shape[0]).reshape(-1, 1)), axis=1)
+        return arr
+
+    def _check_inputs(self, X: np.ndarray, y: np.ndarray, k: int):
+        if type(self.input_graph) is not bool:
+            raise TypeError("input_graph_ must be of type bool.")
+        # check X
+        if not isinstance(X, np.ndarray):
+            raise TypeError("X must be of type np.ndarray.")
+        if not np.issubdtype(X.dtype, np.number):
+            raise TypeError("Adjacency matrix or embedding should have numeric type")
+        elif np.ndim(X) != 2:
+            raise IndexError("Adjacency matrix or embedding must have dim 2")
+        elif not self.input_graph:
+            # embedding was provided
+            if X.shape[1] > X.shape[0]:
+                raise IndexError("dim 1 of an embedding should be smaller than dim 0.")
+        elif X.shape[0] != X.shape[1]:
+            raise IndexError("Adjacency Matrix should be square.")
+
+        # check y
+        if not isinstance(y, np.ndarray):
+            raise TypeError("y must be of type np.ndarray")
+        elif not np.issubdtype(y.dtype, np.integer):
+            raise TypeError("y must have dtype int")
+        elif np.ndim(y) > 2 or (y.ndim == 2 and y.shape[1] > 2):
+            raise IndexError("y must have shape (n) or (n, 1) or (n, 2).")
+        # check k
+        if k is not None and type(k) is not int:
+            raise TypeError("k must be an integer")
+        elif k is not None and k <= 0:
+            raise ValueError("k must be greater than 0")
+
+        if self.input_graph:
+            if not isinstance(self.embedder, BaseSpectralEmbed) and not isinstance(
+                self.embedder, str
+            ):
+                raise TypeError(
+                    "embedder must be either of type str or BaseSpectralEmbed"
+                )
+
+    def _embed(self, X: np.ndarray):
+        # Embed graph if embedding not provided
+        if self.input_graph:
+            if isinstance(self.embedder, BaseSpectralEmbed):
+                embedder = self.embedder
+            elif self.embedder == "ASE":
+                embedder = ase()
+            elif self.embedder == "LSE":
+                embedder = lse()
+            else:
+                raise ValueError(
+                    "Requested embedding method does not exist, if str is passed must "
+                    "be either 'ASE' or 'LSE'."
+                )
+            self.embedding_ = embedder.fit_transform(X)
+        else:
+            self.embedding_ = X
+
+    def fit(
+        self,
+        X: np.ndarray,
+        y: np.ndarray,
+        k: int = None,
```

Comment:
```suggestion
        n_neighbors: int = None,
```
try to have parameters and attributes use the same names as their counterparts in sklearns NearestNeighbors